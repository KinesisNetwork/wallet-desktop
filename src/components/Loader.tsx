import * as React from 'react'

export class Loader extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <div className="sk-folding-cube">
        <div className="sk-cube1 sk-cube" />
        <div className="sk-cube2 sk-cube" />
        <div className="sk-cube3 sk-cube" />
        <div className="sk-cube4 sk-cube" />
      </div>
    )
  }
}
