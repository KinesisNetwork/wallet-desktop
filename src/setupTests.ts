import { configure } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

export default configure({ adapter: new Adapter() })
