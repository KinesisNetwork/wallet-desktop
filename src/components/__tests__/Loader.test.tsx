import { shallow } from 'enzyme'
import * as React from 'react'

import { Loader } from '@components/Loader'
import '../../setupTests'

describe('Loader', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Loader />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders four cubes', () => {
    const wrapper = shallow(<Loader />)

    expect(wrapper.hasClass('sk-folding-cube')).toBe(true)
    expect(wrapper.children()).toHaveLength(4)
  })
})
