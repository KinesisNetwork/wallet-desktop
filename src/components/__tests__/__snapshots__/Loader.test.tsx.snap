// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders correctly 1`] = `
<div
  className="sk-folding-cube"
>
  <div
    className="sk-cube1 sk-cube"
  />
  <div
    className="sk-cube2 sk-cube"
  />
  <div
    className="sk-cube3 sk-cube"
  />
  <div
    className="sk-cube4 sk-cube"
  />
</div>
`;
